src/plugins/image/src/main/js/core/Utils.js

```diff
@@ -201,7 +201,7 @@ define(
      * @return {string}
      */
     var internalPathToRenderFileURL = function (path) {
-      return path ? 'CONTEXT_PATH/render/file.act?path=' + encodeURI(path) : '';
+      return path ? 'CONTEXT_PATH/render/file.act?path=' + encodeURIComponent(path) : '';
```

Comment:
This fixes the rendering in the WYSIWYG and allows the image source in the image menu to not break (where the previously chosen image was not found).

The problem with rendering on the Cascade side, is that the link reads `xyz+folder\image.png` instead of `xyz%2Bfolder\image.png` which will allow the images to render in the browser (just testing using dev tools).

The link gets encoded properly with the `%2B` up until the LookupBank logic starts coming in and the link reverts back to using `+` because the LookupBankResponses uses `getPath` for the link (if I'm understanding that logic correctly).

 I think something similar needs to be done on the Cascade side but I'm not sure where the best place for the logic is.  I was thinking maybe resetting the path [here](https://github.com/hannonhill/Cascade/blob/15b713b8d96c765b87b6278dfd94ca2f1104ee61/src/java/com/hannonhill/cascade/model/render/page/AttributeRewriter.java#L314-L325) to some type of encoded URI Component equivalent in Java (or maybe a simple string replacement) instead of using the `assetPath`. I'm just not sure if this is the appropriate area for this or maybe I'm missing something where it doesn't need to be "re-encoded" with `the 2%B` replacing `+` again.

Reply:
I think switching to `encodeURIComponent` makes sense, because it is specifically meant for encoding query parameters whereas `encodeURI` is meant for encoding an entire URL. My vote would be to switch to this on the TinyMCE side.

Reply:
OK, merging on the TinyMCE & will continue scoping to fix the issue on the server-side.